jetbrains-core/src/software/aws/toolkits/jetbrains/services/ecs/ContainerActions.kt

```diff
@@ -167,7 +169,10 @@ class ExecuteCommandInShellAction(
             if (EcsExecUtils.ensureServiceIsInStableState(project, container.service)) {
                 SessionManagerPluginInstallationVerification.requiresSessionManager(project) {
                     runInEdt {
-                        OpenShellInContainerDialog(project, container, CurrentConnectionSettings.getConnectionSettings(project)).show()
+                        val connectionSettings = AwsConnectionManager.getInstance(project).connectionSettings()
```

Comment:
this is basically the same as the above - and I suspect we use something similar to this in quite a number of places. Would it be worth introducing a new extension function on `Project` something like `fun <T> Project.withAwsConnection(block: (ConnectionSettings) -> T) : T`

which we could then use like this:

```
project.withAwsConnection {
  OpenShellInContainerDialog(project, container, it).show()
}
```

Reply:
The extension function would always return ConnectionSettings right?

Reply:
I'd say its return type would be `T` where `T` is whatever type is returned by the passed `block` - and it is `T` not `T?` because the `withAwsConnection` method would through an exception if the connection wasn't valid - this should only be used in cases where the connection should be valid.

Reply:
Added the function